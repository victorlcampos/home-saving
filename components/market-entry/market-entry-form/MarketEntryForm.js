import React from 'react';
import { StyleSheet, View, Button } from 'react-native';
import t from 'tcomb-form-native';
import _ from 'lodash';

const MarketEntry = t.struct({
  name: t.String,
});

const Form = t.form.Form;

export class MarketEntryForm extends React.Component {
  handleSubmit = () => {
    const id = _.uniqueId('id');
    const value = {
      id,
      ...this._form.getValue(),
    };
    const state = this.props.navigation.state;

    if (value !== null) {
      state.params.addItem(value)();
      this.props.navigation.pop();
    }
  }

  render() {
    const {entries} = this.props;

    return (
      <View style={styles.container}>
        <Form
          ref={c => this._form = c}
          type={MarketEntry}
        />
        <Button
          title="Save"
          onPress={this.handleSubmit}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 15,
    backgroundColor: '#ffffff',
  },
});
