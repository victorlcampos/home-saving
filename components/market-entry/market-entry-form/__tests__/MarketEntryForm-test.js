import React from 'react';
import { MarketEntryForm } from '../MarketEntryForm';
import renderer from 'react-test-renderer';

it('render form', async () => {
  const tree =  renderer.create(<MarketEntryForm />).toJSON();
  expect(tree).toMatchSnapshot();
});
