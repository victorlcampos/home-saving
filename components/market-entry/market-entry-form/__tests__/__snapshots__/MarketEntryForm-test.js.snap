// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render form 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "padding": 15,
    }
  }
>
  <View
    style={Object {}}
  >
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontSize": 17,
            "fontWeight": "500",
            "marginBottom": 7,
          }
        }
      >
        Name
      </Text>
      <View
        style={Object {}}
      >
        <TextInput
          accessibilityLabel="Name"
          allowFontScaling={true}
          autoCapitalize={undefined}
          autoCorrect={undefined}
          autoFocus={undefined}
          blurOnSubmit={undefined}
          clearButtonMode={undefined}
          clearTextOnFocus={undefined}
          editable={undefined}
          enablesReturnKeyAutomatically={undefined}
          keyboardAppearance={undefined}
          keyboardType={undefined}
          maxLength={undefined}
          multiline={undefined}
          numberOfLines={undefined}
          onBlur={undefined}
          onChange={undefined}
          onChangeText={[Function]}
          onContentSizeChange={undefined}
          onEndEditing={undefined}
          onFocus={undefined}
          onKeyPress={undefined}
          onLayout={undefined}
          onSelectionChange={undefined}
          onSubmitEditing={undefined}
          placeholder={undefined}
          placeholderTextColor={undefined}
          returnKeyType={undefined}
          secureTextEntry={undefined}
          selectTextOnFocus={undefined}
          selectionColor={undefined}
          selectionState={undefined}
          style={
            Object {
              "borderColor": "#cccccc",
              "borderRadius": 4,
              "borderWidth": 1,
              "color": "#000000",
              "fontSize": 17,
              "height": 36,
              "marginBottom": 5,
              "paddingHorizontal": 7,
              "paddingVertical": 7,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Save
      </Text>
    </View>
  </View>
</View>
`;
